src/reactfire.js

```diff
@@ -23,12 +38,20 @@ var ReactFireMixin = {
   /*************/
   /* Creates a binding between Firebase and the inputted bind variable as an array */
   bindAsArray: function(firebaseRef, bindVar, cancelCallback) {
-    this._bind(firebaseRef, bindVar, cancelCallback, true);
+    if (!this.setState){
```

Comment:
This check tests how the method is being used. If it is statically like in my new syntax:
```
var ExampleComponent = React.createClass({
  mixins: [ReactFireMixin.bindAsArray(firebaseRef,"propname","cancel")],
});
```

...then `this` doesn't point to a component instance (it doesn't have a `setState` method), and what we want is to return a mixin object with a `componentWillMount` method that does the `_bind` call.

If `this` has a `setState` method then it is a component instance, which means the firebase method is being called in the usual way:

```
var ExampleComponent = React.createClass({
  mixins: [ReactFireMixin],
  componentWillMount: function(){
    this.bindAsArray(firebaseRef,"propname",this.cancel);
  }
});
```

I agree that it tastes a bit hacky, but we need to test somehow whether or not we're currently being used as a static method or an instance method. 

We could of course also define these as two compeltely different methods, but I felt this way was easier since it means the fewest changes to the existing codebase.

I don't know of any examples in React that employs the same approach (I call it "mixin factories", dunno if there's a more academically correct name).